### PR TITLE
Add GetScopeName(id) method to ScopeIdProvider

### PR DESCRIPTION
**Pull request by akopich**
Every id represents a group, while every group
needs a name to be shown under Live tab.
This adds the needed functionality to
ScopeIdProvider.

Tests: Unit tests
Bug: http://b/226889590

### CODE REVIEW

File: src/ClientData/ScopeIdProvider.cpp [hunk 1]
```diff
@@ -26,10 +30,20 @@ namespace orbit_client_data {
...
                 [](const auto& a, const auto& b) { return a.function_id() < b.function_id(); })
                 ->function_id();
-  return std::unique_ptr<NameEqualityScopeIdProvider>(new NameEqualityScopeIdProvider(max_id + 1));
+
+  absl::flat_hash_map<uint64_t, std::string> scope_id_to_name;
+  std::transform(std::begin(instrumented_functions), std::end(instrumented_functions),
+                 std::inserter(scope_id_to_name, std::end(scope_id_to_name)),
+                 [](const auto& instrumented_function) {
+                   return std::make_pair(instrumented_function.function_id(),
+                                         llvm::demangle(instrumented_function.function_name()));
```
Comment (florian-kuebler) [inline src/ClientData/ScopeIdProvider.cpp:39]: There should be no need to demangle the function. By now, we only should have demangled names.

**Comment by akopich** [comment]
I've messed up branching. Closing.

**Review by florian-kuebler — COMMENTED** [review]
Besides the redundant demangling, lgtm.

**Comment by florian-kuebler** [comment]
Oh, just noticed that you already closed the PR...

**Pull request closed without merging**